### ISSUE TITLE
Import local pbix to power bi services

### ISSUE THREAD

**Issue by ManonGirard26** [andrewvillazon/pbipy#37]
Thank you for your library.

Usefull for us but we need to publish a new version of local .pbix file to power bi workspace 
We develop many power bi reports with power bi desktop and want to automate the publishing on cloud.
Using automation with Azure DevOps
Ref: https://learn.microsoft.com/en-us/rest/api/power-bi/imports/post-import

**Comment by andrewvillazon** [comment]
This should be possible to do. Will try look at this in the next couple of weeks.

**Comment by baruchoxman** [comment]
+1 on this.
In the meantime, I've implemented this as an extension class over pbipy, feel free to adopt this code with adjustments:

```
class Import:

    _REPR = [
        "group_id",
    ]

    def __init__(
        self,
        session: requests.Session,
        group_id: str,
    ) -> None:
        self.group_id = group_id
        self.session = session
        self.resource_path = f"/groups/{self.group_id}/imports"
        self.base_path = f"{self.BASE_URL}{self.resource_path}"

    def post_import_in_group(
        self,
        name: str,
        pbix_file_content: io.BytesIO,
        folder_id: str | None = None,
    ) -> dict:
        """
        Post an import pbix file in group for a Power BI workspace.

        Parameters
        ----------
        `name` : `str`
            The name of the pbix model.
        `pbix_file_content` : `io.BytesIO`
            The pbix file content.
        `workspace_id` : `str`
            The workspace id.
        `folder_id` : `str`, optional
            The folder id.

        Returns
        -------
        `dict`
            The response from the Power BI API.
        """

        files = {
            "file": (
                f"{name}.pbix",
                pbix_file_content,
                "application/octet-stream",
            ),
        }
        params = {
            "datasetDisplayName": name,
            "nameConflict": "CreateOrOverwrite",
            "overrideModelLabel": "false",
            "overrideReportLabel": "false",
            "skipReport": "true",
        }
        if folder_id is not None:
            params["subfolderObjectId"] = folder_id

        raw = post_raw(
            self.base_path,
            self.session,
            params=params,
            payload=None,
            files=files,
        )

        return raw
```

**Comment by andrewvillazon** [comment]
Thanks for this, that's helpful. It's nearly there, this end point seems to be doing quite a lot and not much documentation to go on.

**Comment by andrewvillazon** [comment]
@ManonGirard26 @baruchoxman 

I've implemented this as well as methods for the endpoints associated with [Post Import](https://learn.microsoft.com/en-us/rest/api/power-bi/imports/post-import). Feel free to take a look/test drive by installing the branch here.

```
pip install git+https://github.com/andrewvillazon/pbipy@37-import-local-pbix-to-power-bi-services
```

The importing of a file is done from the `PowerBI` client object, using a method named `import_file`, e.g., `pbi.import_file(...)`. As `import` is a reserved word in Python, the method name couldn't reflect the associated action name in PBI.

Let me know if you encounter any issues. If it's good, I'll merge and release.

**Comment by baruchoxman** [comment]
> @ManonGirard26 @baruchoxman
> 
> I've implemented this as well as methods for the endpoints associated with [Post Import](https://learn.microsoft.com/en-us/rest/api/power-bi/imports/post-import). Feel free to take a look/test drive by installing the branch here.
> 
> ```
> pip install git+https://github.com/andrewvillazon/pbipy@37-import-local-pbix-to-power-bi-services
> ```
> 
> The importing of a file is done from the `PowerBI` client object, using a method named `import_file`, e.g., `pbi.import_file(...)`. As `import` is a reserved word in Python, the method name couldn't reflect the associated action name in PBI.
> 
> Let me know if you encounter any issues. If it's good, I'll merge and release.

Hi @andrewvillazon, one feedback on the interface of import_file: I'd be happy to be able to pass a file content (bytes / io.BytesIO) as an input to this function, rather than the path, as I'm creating the file content in memory, and it does not have a path on the disk. Would it be possible to adjust it? Thank you!

**Comment by andrewvillazon** [comment]
@baruchoxman - Thanks for suggesting, hadn't considered this. Have changed the interface, `file_path` has been replaced with `filepath_or_filelike` and should be able to accept a path or any 'file-like'.

```python
pip install git+https://github.com/andrewvillazon/pbipy@37-import-local-pbix-to-power-bi-services
```

**Comment by andrewvillazon** [comment]
Have also removed large file handling from `import_file` and implemented a similar `import_large_file` method to give library users choice on how they want to import large files.

**Comment by baruchoxman** [comment]
@andrewvillazon Tested and it works well for me. Looking forward for the merge and an official release, to update our dependency.
Thank you!

**Comment by andrewvillazon** [comment]
Merged and released under: [v2.10.0](https://github.com/andrewvillazon/pbipy/releases/tag/v2.10.0)
